tests/cross-language/go/encrypt_test.go

```diff
@@ -0,0 +1,123 @@
+package cltf
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/cucumber/godog"
+	"github.com/go-sql-driver/mysql"
+	"github.com/godaddy/asherah/go/appencryption"
+	"github.com/godaddy/asherah/go/appencryption/pkg/crypto/aead"
+	"github.com/godaddy/asherah/go/appencryption/pkg/kms"
+	"github.com/godaddy/asherah/go/appencryption/pkg/persistence"
+	"github.com/pkg/errors"
+)
+
+var (
+	payloadString          string
+	encryptedPayloadString string
+	connection             *sql.DB
+)
+
+func iHave(payload string) error {
+	payloadString = payload
+
+	return nil
+}
+
+func iEncryptTheData() error {
+	crypto := aead.NewAES256GCM()
+	manager, _ := kms.NewStatic(KeyManagementStaticMasterKey, crypto)
+	metastore := persistence.NewSQLMetastore(connection)
+	policy := appencryption.NewCryptoPolicy()
+	config := &appencryption.Config{
+		Service: DefaultServiceID,
+		Product: DefaultProductID,
+		Policy:  policy,
+	}
+
+	factory := appencryption.NewSessionFactory(config, metastore, manager, crypto)
+	defer factory.Close()
+
+	sess, err := factory.GetSession(DefaultPartitionID)
+	if err != nil {
+		panic(err)
+	}
+	defer sess.Close()
+
+	dataRow, e := sess.Encrypt([]byte(payloadString))
+	if e != nil {
+		panic(e)
+	}
+
+	encryptedData, _ := json.Marshal(dataRow)
+	encryptedPayloadString = base64.StdEncoding.EncodeToString(encryptedData)
+
+	return nil
+}
+
+func ishouldgetencryptedData() error {
+	var filePath = fmt.Sprintf("%s%s", FileDirectory, FileName)
+
+	if _, err := os.Stat(filePath); err == nil {
+		os.Remove(filePath)
+	}
+
+	f, err := os.Create(filePath)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	f.WriteString(encryptedPayloadString)
+
+	return nil
+}
+
+func encryptedDatashouldnotbeequaltodata() error {
+	if payloadString == encryptedPayloadString {
+		return errors.New("Encryption failure")
+	}
+
+	return nil
+}
+
+// nolint: deadcode
```

Comment:
Is this needed? Seems odd that an exported function would be considered unused.

Reply:
Was able to run everything successfully without needing this